subprojects/core-api/src/main/java/org/gradle/internal/scripts/DefaultScriptFileResolver.java

```diff
@@ -15,18 +15,23 @@
  */
 package org.gradle.internal.scripts;
 
+import org.gradle.scripts.ScriptingLanguage;
+
 import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 
 import static java.util.Collections.emptyList;
+
 import static org.gradle.internal.FileUtils.hasExtension;
 
 public class DefaultScriptFileResolver implements ScriptFileResolver {
 
+    private static final String[] EXTENSIONS = scriptingLanguageExtensions();
```

Comment:
Why this copy?

Reply:
It's not exactly a copy, it's a precomputation of all `ScriptingLanguage` extensions to avoid the `ScriptingLanguage.all().get(i).getExtension()` indirections on every request.

Reply:
Right, I misread that.